module1/homework/lengthSort/sort.cpp

```diff
@@ -1,14 +1,20 @@
 #include "sort.hpp"
+#include <algorithm>
 
 std::deque<std::string> lengthSort(std::forward_list<std::string>& forwardList) {
-    std::deque<std::string> dq;
-    auto it = dq.begin();
 
-    for (const auto& word : forwardList) {
-        for (it = dq.begin(); it != dq.end(); it++)
-            if (it->size() > word.size() || it->size() == word.size() && *it > word)
-                break;
-        dq.insert(it, word);
+    std::deque<std::string> dq;
+    
+    forwardList.sort( [](const auto str1, const auto str2){
+        if(str1.size() == str2.size()){
+            return str1 < str2;
+        }
+        return str1.size() < str2.size();
+    } );
+    
+    for(const auto el : forwardList){
+        dq.push_back(el);
     }
```

Comment:
you can even do something like that:
```cpp
std::deque<std::string> dq{forwardList.begin(), forwardList.end()};
```
It will be already initialized with the content of forwardList, so there will be no need a this loop.